app/addons/documents/changes/components.react.jsx

```diff
@@ -0,0 +1,244 @@
+    },
+
+    getInitialState: function () {
+      return this.getStoreState();
+    },
+
+    getFilters: function () {
+      return _.map(this.state.filters, function (filter) {
+        return <Filter key={filter} label={filter} />;
+      }, this);
+    },
+
+    render: function () {
+      var filters = this.getFilters();
+
+      return (
+        <div className="tab-content">
+          <div className="tab-pane active" ref="filterTab">
+            <div className="changes-header js-filter">
+              <AddFilterForm tooltip={this.props.tooltip} filter={this.state.filter} onSubmit={this.submitForm} />
+              <ul className="filter-list">{filters}</ul>
+            </div>
+          </div>
+        </div>
+      );
+    }
+  });
+
+
+  var AddFilterForm = React.createClass({
+    getInitialState: function () {
+      return {
+        filter: ''
+      };
+    },
+
+    getDefaultProps: function () {
+      return {
+        tooltip: ''
+      };
+    },
+
+    submitForm: function (e) {
+      e.preventDefault();
+      e.stopPropagation();
+      if (_.isEmpty(this.state.filter)) {
+        return;
+      }
+      Actions.addFilter(this.state.filter);
+    },
+
+    componentDidMount: function () {
+      this.focusFilterField();
+    },
+
+    componentDidUpdate: function () {
+      this.focusFilterField();
+    },
+
+    focusFilterField: function () {
+      this.refs.addItem.getDOMNode().focus();
+    },
+
+    onChangeFilter: function (e) {
+      this.setState({ filter: e.target.value });
+    },
+
+    render: function () {
+      return (
+        <form className="form-inline js-filter-form" onSubmit={this.submitForm}>
+          <fieldset>
+            <input type="text" ref="addItem" className="js-changes-filter-field" placeholder="Type a filter"
+              onChange={this.onChangeFilter} value={this.state.filter} />
+            <button type="submit" className="btn btn-primary">Filter</button>
+            <div className="help-block">
+              <strong ref="helpText">e.g. _design or document ID</strong>
+              {' '}
+              <FilterTooltip tooltip={this.props.tooltip} />
+            </div>
+          </fieldset>
+        </form>
+      );
+    }
+  });
+
+
+  var FilterTooltip = React.createClass({
+    componentDidMount: function () {
+      if (this.props.tooltip) {
+        $(this.refs.tooltip.getDOMNode()).tooltip();
+      }
+    },
+
+    render: function () {
+      if (!this.props.tooltip) {
+        return false;
+      }
+      return (
+        <i ref="tooltip" className="icon icon-question-sign js-filter-tooltip" data-toggle="tooltip"
+          data-original-title={this.props.tooltip}></i>
+      );
+    }
+  });
+
+
+  var Filter = React.createClass({
+    propTypes: {
+      label: React.PropTypes.string.isRequired
+    },
+
+    remove: function (e) {
+      e.preventDefault();
+      Actions.removeFilter(this.props.label);
```

Comment:
The `<Filter />` component should be "dumb" and not call an action directly but call a callback passed to it from its controller component.